frontend/app/components/ToolCard.tsx

```diff
@@ -27,6 +29,45 @@ interface ToolCardProps {
 export default function ToolCard (props: ToolCardProps) {
   const { tool, onToolUpdate } = props;
 
+  // ToolCard action handlers with google analytics tracking
+  const analyticsConsentContext = useContext(AnalyticsConsentContext);
+  const sendAnalyticsEvent = async (category: string, action: string, label: string, value: number) => {
+    if (analyticsConsentContext && analyticsConsentContext.analyticsConsentGiven) {
+      await ReactGA.event({category, action, label, value});
+    }
+  };
+  const handleMoreInfoClick = async (tool: Tool) => {
+    try {
+      if (!showMoreInfo) { // On opening more info
+        await sendAnalyticsEvent('Tool Info', 'More Info Clicked', tool.name, tool.canvas_id);
```

Comment:
can we also track the Canvas course_id? maybe combine canvas course id with tool.canvas_id?